dist/user_install_script.sh

```diff
@@ -0,0 +1,163 @@
+            yes "Decky!" | passwd deck
+            trap temp_pass_cleanup EXIT # make sure password is removed when application closes
+            PASS="Decky!"
+        else exit 1; fi
+    else
+        # get password
+        FINISHED="false"
+        while [ "$FINISHED" != "true" ]; do
+            PASS=$(zenity --title="Decky Installer" --width=300 --height=100 --entry --hide-text --text="Enter your admin password")
+            if [[ $? -eq 1 ]] || [[ $? -eq 5 ]]; then
+                exit 1
+            fi
+            if ( echo "$PASS" | sudo -S -k true ); then
+                FINISHED="true"
+            else
+                zenity --title="Decky Installer" --width=150 --height=40 --info --text "Incorrect Password"
+            fi
+        done
+    fi
+
+    if ! [ $USER = "deck" ]; then
+        zenity --title="Decky Installer" --width=300 --height=100 --warning --text "You appear to not be on a deck.\nDecky should still mostly work, but you may not get full functionality."
+    fi
+
+    # get user dir before rerunning as root, otherwise it'll just be 'home/root'
+    USER_DIR="$(getent passwd $USER | cut -d: -f6)"
+    HOMEBREW_FOLDER="${USER_DIR}/homebrew"
+    echo "$PASS" | sudo -S -k sh "$0" "$USER_DIR" "$HOMEBREW_FOLDER" # rerun script as root
+    exit 1
+fi
+
+# all code below should be run as root
+USER_DIR=$1
+HOMEBREW_FOLDER=$2
+
+# if decky is already installed, then also add an 'uninstall' prompt
+if [[ -f "${USER_DIR}/homebrew/services/PluginLoader" ]] ; then
+    BRANCH=$(zenity --title="Decky Installer" --width=360 --height=170 --list --radiolist --text "Select Option:" --hide-header --column "Buttons" --column "Choice" --column "Info" TRUE "release" "(Recommended option)" FALSE "prerelease" "(May be unstable)" FALSE "uninstall decky loader" "")
+else
+    BRANCH=$(zenity --title="Decky Installer" --width=300 --height=100 --list --radiolist --text "Select Branch:" --hide-header --column "Buttons" --column "Choice" --column "Info" TRUE "release" "(Recommended option)" FALSE "prerelease" "(May be unstable)" )
+fi
+if [[ $? -eq 1 ]] || [[ $? -eq 5 ]]; then
+    exit 1
+fi
+
+# uninstall if uninstall option was selected
+if [ "$BRANCH" == "uninstall decky loader" ] ; then
+    (
+    echo "35" ; echo "# Disabling and removing services" ;
+    sudo systemctl disable --now plugin_loader.service > /dev/null
+    sudo rm -f "${USER_DIR}/.config/systemd/user/plugin_loader.service"
+    sudo rm -f "/etc/systemd/system/plugin_loader.service"
+
+    echo "70" ; echo "# Removing Temporary Files" ;
+    rm -rf "/tmp/plugin_loader"
+    rm -rf "/tmp/user_install_script.sh"
+
+    echo "100" ; echo "# Cleaning services folder" ;
+    sudo rm "${HOMEBREW_FOLDER}/services/PluginLoader"
+    ) |
+    zenity --progress \
+  --title="Decky Installer" \
+   --width=300 --height=100 \
+  --text="Installing..." \
```

Comment:
Shouldn't this say uninstalling

Reply:
Yep, although it will probably never actually show up as the first command run replaces it